content/blog/migrating-a-cloud-application-to-kubernetes/index.md

```diff
@@ -0,0 +1,391 @@
+            spec: {
+                containers: [{
+                    name: databaseAppName,
+                    image: awsx.ecr.buildAndPushImage("database-side-service", "./databaseside").image(),
+                    ports: [{ name: "http", containerPort: 5432 }],
+                    env: [
+                        { name: "ADMIN_NAME", value: sqlAdminName },
+                        { name: "ADMIN_PASSWORD", value: sqlAdminPassword },
+                    ],
+                    volumeMounts: [{
+                        name: "persistent-volume",
+                        mountPath: "/persistentVolume"
+                    }],
+                    resources: {
+                        limits: {
+                            memory: "1Gi",
+                            cpu: "1000m"
+                        }
+                    }
+                }],
+                volumes: [{
+                    name: "persistent-volume",
+                    awsElasticBlockStore: {
+                        volumeID: ebsVolume.id,
+                    },
+                }],
+                affinity: {  
+                    nodeAffinity: {
+                        requiredDuringSchedulingIgnoredDuringExecution: {
+                            nodeSelectorTerms: [{
+                                matchExpressions: [{
+                                    key: "failure-domain.beta.kubernetes.io/zone",
+                                    operator: "In",
+                                    values: [ebsVolume.availabilityZone],
+                                }]
+                            }]
+                        }
+                    }
+                }
+            }
+        },
+        strategy: {
+            type: "Recreate"
+        }
+    }}, {
+    deleteBeforeReplace: true,
+    provider: eksCluster.provider,
+});
+```
+
+In order to have a a way for the pods in cluster to communicate with our database, we create a Kubernetes service for it. It is best practice to keep databases and other similar components entirely within the cluster, and not allow them to be accessed directly the internet.
+
+```typescript
+const databasesideListener = new k8s.core.v1.Service("database-side-listener", {
+    metadata: { labels: databaseDeployment.metadata.labels },
+    spec: {
+        type: "ClusterIP",
+        ports: [{ port: 5432, targetPort: "http" }],
+        selector: databaseAppLabels,
+        publishNotReadyAddresses: false,
+    }}, {
+    provider: eksCluster.provider,
+    }
+);
+
+const postgresqlAddress = databasesideListener.spec.clusterIP;
+});
+```
+
+Now, all that remains is deploying the client and server containers to Kubernetes. Since the PERN application already deployed them as docker containers, it doesn't take much effort to reconfigure it for Kubernetes. Unlike the database, these components can safely be opened up to the internet using a service with a Load Balancer type.
```

Comment:
```suggestion
Now, all that remains is deploying the client and server containers to Kubernetes. Since the PERN application already deployed them as docker containers, it doesn't take much effort to reconfigure it for Kubernetes. Unlike the database, these components can safely be open to the Internet using a service with a Load Balancer type.
```